### PR TITLE
Fix memleaks in async api

### PR DESCRIPTION
**Pull request by bernd-edlinger**
Fixes: #5950

### CODE REVIEW

**Comment by bernd-edlinger** [comment]
The idea is mainly to do OPENSSL_init on external API, and not on internal APIs

File: crypto/async/async.c [hunk 1]
```diff
@@ -369,32 +368,41 @@ int ASYNC_init_thread(size_t max_size, size_t init_size)
...
+
+    async_delete_thread_state();
 }
 
 ASYNC_JOB *ASYNC_get_current_job(void)
 {
     async_ctx *ctx;
 
+    if (!OPENSSL_init_crypto(OPENSSL_INIT_ASYNC, NULL))
+        return 0;
```
Comment (stevelinsell) [inline crypto/async/async.c]: Should return NULL rather than 0 as it's returning a pointer.

Comment (bernd-edlinger) [reply crypto/async/async.c]: Yes.  copied from somewhere else...

**Comment by bernd-edlinger** [comment]
Merged to master as 74a8acbdfb2c7f398d1ae2fe914cd32b437f6df4
and 1.1.0 as ba1cf1b

Thanks!

**Pull request closed without merging**